components/style/properties/longhand/inherited_text.mako.rs

```diff
@@ -737,7 +737,143 @@ ${helpers.single_keyword("text-align-last",
+
+        #[derive(Debug, Clone, PartialEq, Copy)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct Keyword {
+            pub fill: bool,
+            pub shape: super::ShapeKeyword
+        }
+    }
+
+    #[derive(Debug, Clone, PartialEq, Copy)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct SpecifiedValue (pub Option<Keyword>);
+
+    impl ToCss for computed_value::T {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            match self.0 {
+                Some(keyword) => keyword.to_css(dest),
+                None => dest.write_str("none"),
+            }
+        }
+    }
+    impl ToCss for SpecifiedValue {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            match self.0 {
+                Some(keyword) => keyword.to_css(dest),
+                None => dest.write_str("none"),
+            }
+        }
+    }
+
+    #[derive(Debug, Clone, PartialEq, Copy)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct Keyword {
+        pub fill: Option<bool>,
+        pub shape: Option<ShapeKeyword>
+    }
+
+    impl ToCss for computed_value::Keyword {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if self.fill {
+                try!(dest.write_str("filled"));
+            } else {
+                try!(dest.write_str("open"));
+            }
+            try!(dest.write_str(" "));
+            self.shape.to_css(dest)
+        }
+    }
+
+    impl ToCss for Keyword {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if let Some(fill) = self.fill {
+                if fill {
+                    try!(dest.write_str("filled"));
+                } else {
+                    try!(dest.write_str("open"));
+                }
+            }
+            if let Some(shape) = self.shape {
+                try!(dest.write_str(" "));
+                try!(shape.to_css(dest));
+            }
+            Ok(())
+        }
+    }
+
+    define_css_keyword_enum!(ShapeKeyword:
+                             "dot" => Dot,
+                             "circle" => Circle,
+                             "double-circle" => DoubleCircle,
+                             "triangle" => Triangle,
+                             "sesame" => Sesame);
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T(None)
+    }
+
+    impl ToComputedValue for SpecifiedValue {
+        type ComputedValue = computed_value::T;
 
+        #[inline]
+        fn to_computed_value(&self, _context: &Context) -> computed_value::T {
+            match self.0 {
+                Some(keyword) => computed_value::T(Some(computed_value::Keyword {
+                    fill: keyword.fill.unwrap_or(true),
+                    // FIXME: How to check writing mode? Circle in vertical writing mode, and sesame in horizontal.
```

Comment:
https://doc.servo.org/style/values/computed/struct.Context.html , `.style().writing_mode.is_vertical()` or something like that. Not sure if you're supposed to use the inherited one, @SimonSapin ? Perhaps we intersect the regular one with the inherited one?